### PR TITLE
Actions: Add a mingw-w64+ffmpeg & Remove appveyor

### PR DESCRIPTION
**Pull request by 1480c1**
# Description

Moves over the appveyor job to github actions

Tests mingw-w64 compilation along building FFmpeg with libdav1d as well

the resulting ffmpeg.exe binary is uploaded too.

@ePirat @luigi311 @EwoutH can you guys help review this?

# Issue

# Author(s)
@1480c1 

# Performance impact

- [ ] quality
- [ ] memory
- [ ] speed
- [ ] 8 bit
- [ ] 10 bit
- [x] N/A

# Test set
- [ ] obj-1-fast can be found [here](https://media.xiph.org/video/derf/objective-1-fast.tar.gz)
- [ ] other
- [x] N/A

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

**Comment by EwoutH** [comment]
Nice work, LGTM

**Comment by EwoutH** [comment]
One comment: Can we add a deploy function to add the artifact to the release page, like AppVeyor did?

File: .github/workflows/mingw-w64.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    runs-on: windows-2019
+    steps:
+      - name: Clone SVT-AV1
+        uses: actions/checkout@v2
+      - name: Clone FFmpeg for caching
+        uses: actions/checkout@v2
+        with:
+          repository: FFmpeg/FFmpeg
+          path: ffmpeg
+          fetch-depth: 1
```
Comment (EwoutH) [inline .github/workflows/mingw-w64.yml:37]: `fetch-depth: 1` is already the default

File: .github/workflows/mingw-w64.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            cat ffbuild/config.log
+            exit 1
+          }
+      - name: Build FFmpeg
+        run: make -j 10
+      - name: Upload ffmpeg (mingw-w64)
+        uses: actions/upload-artifact@v2
+        with:
+          name: svtav1-mingw-w64-ffmpeg
+          path: ffmpeg.exe
```
Comment (ePirat) [inline .github/workflows/mingw-w64.yml:94]: Why is this uploaded? The result will not be usable anyway?

Comment (ePirat) [reply .github/workflows/mingw-w64.yml:94]: Ah, nevermind. I missed it's doing a full static build.

**Comment by 1480c1** [comment]
> One comment: Can we add a deploy function to add the artifact to the release page, like AppVeyor did?

I can try, but it may take some time

File: .github/workflows/mingw-w64.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        run: |
+          Copy-Item $env:GITHUB_WORKSPACE/ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1.patch $HOME
+          pacman -Suu --ask=20 --noconfirm --noprogressbar
+          pacman -Su --ask=20 --noconfirm --noprogressbar
+          pacman -Sy --ask=20 --noconfirm --noprogressbar --needed mingw-w64-x86_64-ccache mingw-w64-x86_64-yasm mingw-w64-x86_64-cmake mingw-w64-x86_64-pkg-config mingw-w64-x86_64-gcc mingw-w64-x86_64-dav1d make
+
+      - name: Configure SVT-AV1
+        run: cmake -B Build -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX=C:/msys64/mingw64 -DBUILD_SHARED_LIBS=OFF
+      - name: Build SVT-AV1
+        run: make -j 4 -C Build install
```
Comment (luigi311) [inline .github/workflows/mingw-w64.yml:74]: Does mingw not have access to nproc?

File: .github/workflows/mingw-w64.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        run: git apply -3 --ignore-whitespace "$HOME/0001-Add-ability-for-ffmpeg-to-run-svt-av1.patch"
+      - name: Configure FFmpeg
+        shell: bash -eo pipefail {0}
+        run: |
+          ./configure --arch=x86_64 --cc="ccache gcc" --cxx="ccache g++" --enable-gpl --enable-libsvtav1 --enable-libdav1d --enable-static --disable-shared || {
+            cat ffbuild/config.log
+            exit 1
+          }
+      - name: Build FFmpeg
+        run: make -j 10
```
Comment (luigi311) [inline .github/workflows/mingw-w64.yml:89]: Why 10 here and 4 in the svt build?

**Comment by hassount** [comment]
so do we remove Appveyor from the repo all together ?

**Comment by 1480c1** [comment]
That is the hope with this PR so that way we reduce the number of external dependencies in terms of CI, I might need to remove it from the readme

**Comment by hassount** [comment]
> That is the hope with this PR so that way we reduce the number of external dependencies in terms of CI, I might need to remove it from the readme.
> 
> Alternatively, we can still keep appveyor as a backup for windows CI if we want to

Agreed, but right now it's failing, can we maybe add the file and skip all branches, remove it from the read me, keep it there for a while until we make sure actions is working properly, then uninstall it from the repo.

**Comment by 1480c1** [comment]
Sure, I can do that

**Pull request merged**